diff --git a/ggml/src/ggml-cpu/ggml-cpu.c b/ggml/src/ggml-cpu/ggml-cpu.c
index 50ca3c8..7d916a9 100644
--- a/ggml/src/ggml-cpu/ggml-cpu.c
+++ b/ggml/src/ggml-cpu/ggml-cpu.c
@@ -1674,7 +1674,11 @@ static void ggml_compute_forward_mul_mat_id(
                 moe_log_state = moe_log_file ? 1 : 0;
             }
             if (moe_log_state == 1 && strstr(src0->name, "ffn_gate_exps")) {
+#if defined(_MSC_VER)
+                _lock_file(moe_log_file);
+#else
                 flockfile(moe_log_file);
+#endif
                 for (int64_t iid1 = 0; iid1 < ids->ne[1]; ++iid1) {
                     fprintf(moe_log_file, "%s", src0->name);
                     for (int id = 0; id < n_ids; ++id) {
@@ -1683,7 +1687,11 @@ static void ggml_compute_forward_mul_mat_id(
                     }
                     fputc('\n', moe_log_file);
                 }
+#if defined(_MSC_VER)
+                _unlock_file(moe_log_file);
+#else
                 funlockfile(moe_log_file);
+#endif
             }
         }
     }
